core/src/main/scala/org/bitcoins/core/wallet/utxo/UTXOSpendingInfo.scala

```diff
@@ -34,10 +34,9 @@ import org.bitcoins.core.script.crypto.HashType
 import org.bitcoins.core.util.CryptoUtil
 
 /**
-  * Contains the information required to spend a unspent transaction output (UTXO)
-  * on a blockchain.
+  * Contains the information required to sign an unspent transaction output (UTXO) for a single key
   */
-sealed abstract class UTXOSpendingInfo {
+sealed abstract class UTXOSpendingInfoSingle {
```

Comment:
My proposal to make this a true ADT: 

```scala
sealed abstract class UTXOSpendingInfo{

  /** The funding transaction's txid and the index of the output in the transaction we are spending */
  def outPoint: TransactionOutPoint

  def amount: CurrencyUnit

  def scriptPubKey: ScriptPubKey

  /** the actual output itself we are spending */
  def output: TransactionOutput = {
    TransactionOutput(value = amount, scriptPubKey = scriptPubKey)
  }

  /** The signer signing in the output above */
  def signer: Sign

  def hashType: HashType

  def redeemScriptOpt: Option[ScriptPubKey]

  def scriptWitnessOpt: Option[ScriptWitness]

  def conditionalPath: ConditionalPath
}

sealed trait UTXOSpendingInfoSingle extends UTXOSpendingInfo

/**
  * Contains the information required to fully sign an unspent transaction output (UTXO)
  * on a blockchain.
  *
  * If you want to partially sign a UTXO you will likely need to use UTXOSpendingInfoSingle.
  */
sealed abstract class UTXOSpendingInfoFull extends UTXOSpendingInfo { ... 
```

This gives us a disjoint type between `UTXOSpendingInfoFull` and `UTXOSpendingInfoSingle` so we can safely pattern match and use the compiler's exhaustiveness checks.